### DISCUSSION TITLE
Extension not working automatically

### DISCUSSION THREAD

**Discussion by heysujal** [open-sauced-craftwork/browser-extension#14] [Q&A] [answered]
I have already made a PR for the work I have done till now. My extension gets the job done of adding a button, but it only works when someone clicks on the extension icon and not automatically. I have tried out almost all the fixes there are on the internet. Most of them are outdated and use manifest V2 but new solutions are also not working for me.
A bit about how my code is working.
I have written my code in content App.jsx which is doing the main job of checking if the user is a member or not and adding a button but before returning this component I call a function that does all these jobs by calling other functions.
How can I get my code to add a button on the page without the need of clicking on the extension icon.

**Comment by anush008** [comment, marked as answer]
Hey Sujal. This could be due to how the extension settings are configured.
* Right-click on your extension and click on "Manage extension".
* On the extension settings page, verify that site access permission for the extension is not set to "On click". If it is, update it as needed.
![ext-perms](https://user-images.githubusercontent.com/46051506/225069140-69a4f57e-c2da-4cc3-bcd0-a317bc5f65ba.png)

Hope this helps.

**Reply by heysujal** [reply]
Hi @Anush008 . I checked that option, it is not set to `On click` . What should I do next?

**Reply by anush008** [reply]
Have you set the `host_permissions` appropriately in the manifest?

**Reply by heysujal** [reply]
@Anush008 "host_permissions": ["<all_urls>"] .This is set in manifest

**Reply by anush008** [reply]
That's weird. If it's not the configuration, re-checking the implementation is all I can think of.

**Reply by heysujal** [reply]
I can tell you how my code is structured, 
in my `main.jsx` I am just returning the `App.jsx`
In my `App.jsx` I am having a `useEffect()` in the component function in which another function `handleLoad ` is called. In the `handleLoad() `, first I query the chrome Tab then a function insertButton() is called using `chrome.scripting.executeScript` .
Inside this `insertButton() `there exists some code for dom manipulation.
My guess is that the `main.jsx` starts rendering only when I click on the icon. It is not run automatically.
I was given a hint in discord by someone to remove the `useEffect` and call the `handleLoad()` directly but it didn't work. When I click on the extension icon it is just a small white box that doesn't even open fully.

**Reply by anush008** [reply]
It seems like the issue is related to the use of the `chrome.scripting.executeScript()` function in the `React.useEffect()`.

To solve this issue, you can try performing the DOM manipulation using a content-script directly.

**Reply by heysujal** [reply]
All of my code is in content script already.But still I will try to make some changes and shift some code to background.js

**Reply by anush008** [reply]
That background workers aren't for DOM manipulation. Your problem still lies in the content-scripts.

**Reply by heysujal** [reply]
@Anush008 Can I dm you on discord? Can you have a look at my code?
heysujal#9726 this is my username on discord.

**Reply by heysujal** [reply]
@Anush008  Can you give your discord handle?

**Reply by anush008** [reply]
@Anush#5038. Here you go.

**Reply by heysujal** [reply]
is it correct? I am not able to find you in discord. Can you send me a message instead ?

**Reply by anush008** [reply]
Sent you a request.
